retworkx-core/src/shortest_path/dijkstra.rs

```diff
@@ -98,24 +138,25 @@ use crate::min_scored::MinScored;
 /// assert_eq!(res.unwrap(), expected_res);
 /// // z is not inside res because there is not path from b to z.
 /// ```
-pub fn dijkstra<G, F, K, E>(
+pub fn dijkstra<G, F, K, E, S>(
     graph: G,
     start: G::NodeId,
     goal: Option<G::NodeId>,
     mut edge_cost: F,
     mut path: Option<&mut DictMap<G::NodeId, Vec<G::NodeId>>>,
-) -> Result<DictMap<G::NodeId, K>, E>
+) -> Result<S, E>
 where
-    G: IntoEdges + Visitable,
-    G::NodeId: Eq + Hash,
+    G: IntoEdges + Visitable + NodeIndexable,
+    G::NodeId: Eq + Hash + IndexType,
```

Comment:
I don't think this is needed and actually it'd restrict the graph types that we can use.

```suggestion
    G::NodeId: Eq + Hash,
```